app/models/user.rb

```diff
@@ -3,5 +3,13 @@ class User < ActiveRecord::Base
 
   has_many :links
 
-  validates_uniqueness_of :email
+  validates :name,  presence: true, length: { maximum: 50 }
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
I might talk about this in class, but validating emails via regex is a no-no. Sort of an unwritten rule.